lib/core/test/unit/Cardano/Wallet/Primitive/ModelSpec.hs

```diff
@@ -390,41 +366,22 @@ prop_availableUTxO
 prop_availableUTxO makeProperty =
     forAllShrink (scale (* 4) genUTxO) shrinkUTxO
         $ \utxo ->
-    forAllShrink (listOf genTxInputs) (shrinkList shrinkTxInputs)
-        $ \pendingTxInputs ->
-    inner utxo pendingTxInputs
+    forAllShrink (scale (`div` 2) $ listOf genTx) (shrinkList shrinkTx)
```

Comment:
Why do we `scale (div 2)` here?

Reply:
Like I think I understand the size paramater, but I'm not sure why we need to divide it by two here.

Reply:
By using `scale (div 2)`, we scale down the size parameter passed to all the inner generators of `genTx`, meaning that "smaller" transactions will be generated.

In testing, I found that this didn't affect the coverage conditions we care about, while making the tests run faster.

Reply:
In general, users of `genTx` can choose the size of transactions to generate by manipulating the size parameter in this way.

It's the same principle for most of the other generators that we provide in `Gen` modules.

If you find that you can reduce the size parameter **_and_** keep the coverage you want (which should be explicitly tested), then it often makes sense to do so, as we can reduce the overall test execution time.

Reply:
We could also have used `scale (mod n)` here, for some value of `n`, which would place a **_fixed_** upper limit on the size of transactions (while still allowing the size to vary).